### ISSUE TITLE
[a11y]: ai-chat does not apply text spacing consistently

### ISSUE THREAD

**Issue by Daniel-Schulz-Private** [carbon-design-system/carbon-labs#909]
### Browser

Chrome

### Operating System

MacOS

### Package version

0.18.0

### React version

18.3.1

### Automated testing tool and ruleset

Stylus

### Description

Text Spacing is not applied for 
- the burger menu of ai-chat (holding only the item clear history at the moment)
- the links under References, i.e. the LinkList element
- Buttons in the Rate a Response dialog and
- Placeholder text

Recommended fix: Text spacing should be applied to all the elements in the page.

User impact: When users override author-specified text spacing to improve their reading experience, there is a loss of content or functionality. This negatively impacts users with low vision or a learning disability.

### WCAG 2.1 Violation

WCAG Mapping: 1.4.12 Text Spacing

### Reproduction/example

n/a

### Steps to reproduce

1. open ai-chat
2. observe missing text spacing of the mentioned items

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-labs/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon-labs/issues) for duplicate problems
